pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the MPSPrep template.
+"""
+
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+
+
+class MPSPrep(Operation):
+    r"""Prepares an initial state from a matrix product state (MPS) representation.
+
+    .. note::
+
+        Currently, this operator can only be used with ``qml.device(“lightning.tensor”)``.
+
+
+    Args:
+        mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
+            product of site matrices. See the usage details section for more information.
+
+        wires (Sequence[int]): wires that the template acts on
+
+    **Example**
+
+    .. code-block::
+
+        mps = [
+            np.array([[0.0, 0.107], [0.994, 0.0]]),
+            np.array(
+                [
+                    [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                    [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                ]
+            ),
+            np.array(
+                [
+                    [[-1.0, 0.0], [0.0, 0.0]],
+                    [[0.0, 0.0], [0.0, 1.0]],
+                    [[0.0, -1.0], [0.0, 0.0]],
+                    [[0.0, 0.0], [1.0, 0.0]],
+                ]
+            ),
+            np.array([[-1.0, -0.0], [-0.0, -1.0]]),
+        ]
+
+        dev = qml.device("lightning.tensor", wires = 3)
+        @qml.qnode(dev)
+        def circuit():
+            qml.MPSPrep(mps, wires = [0,1,2])
+            return qml.state()
+
+    .. code-block:: pycon
+
+        >>> print(circuit())
+        [ 0.    +0.j  0.    +0.j  0.    +0.j -0.1066+0.j  0.    +0.j  0.    +0.j
+          0.    +0.j  0.    +0.j  0.    +0.j  0.    +0.j  0.    +0.j  0.    +0.j
+          0.9943+0.j  0.    +0.j  0.    +0.j  0.    +0.j]
+
+    .. details::
+        :title: Usage Details
```

Comment:
Really nice explanation :rocket: 
I would not have understood without this :muscle: